### PR TITLE
[MLv2] Update types to accommodate `QueryDisplayInfo`

### PR DESCRIPTION
**Pull request by nemanjaglumac**
The progress of https://github.com/metabase/metabase/issues/37389 is blocked by https://github.com/metabase/metabase/issues/37765.

The initial attempt was to both add new types and take care of `isEditable` query methods. In order to unblock further work on #37389, I decided to split the tasks into three. This PR is just the first on the three.

It should unblock the work on migrating `isNative` and `isStructured` methods.

### CODE REVIEW

File: frontend/src/metabase-lib/metadata.ts [hunk 1]
```diff
@@ -143,6 +144,23 @@ declare function DisplayInfoFn(
...
   segment: SegmentMetadata,
 ): SegmentDisplayInfo;
+/**
+ * Even though it seems weird to pass the same query two times,
+ * this function follows the same pattern as the other displayInfo functions.
+ * The first two parameters are always a query, and a stage.
+ * The third parameter is what you would like to have info about.
+ * It just only happens that the thing we're examining is (again) the query itself.
+ *
+ * @example
```
Comment (uladzimirdev) [inline frontend/src/metabase-lib/metadata.ts:154]: I'd remove example as it's TS and you know a signature already.

File: frontend/src/metabase-lib/metadata.ts [hunk 2]
```diff
@@ -143,6 +144,23 @@ declare function DisplayInfoFn(
...
+ * const query = question.query();
+ * const { isEditable, isNative } = Lib.displayInfo(query, -1, query);
+ *
+ */
+declare function DisplayInfoFn(
+  query: Query,
```
Comment (uladzimirdev) [inline frontend/src/metabase-lib/metadata.ts:160]: I'd still name not required things as `_query` and `_stageIndex` to stress out they are optional

**Review by uladzimirdev — COMMENTED** [review]
Please check comments inline

File: frontend/src/metabase-lib/metadata.ts (hunk 2 above)
Comment (nemanjaglumac) [reply frontend/src/metabase-lib/metadata.ts:160]: It seems that I don't understand something here. How are the initial query and stageIndex optional?

`Lib.displayInfo(query)` doesn't give you anything useful
`Lib.displayInfo(query, -1, query)` returns `{ isNative, isEditable }`

image

Comment (uladzimirdev) [reply frontend/src/metabase-lib/metadata.ts:160]: I meant that you could have in JS/TS a function that accepts one parameter - query - but passes down 3 (query, -1, query)

anyhow, I'd not spend time on it, if @kamilmielnik is agreed, I'd go with a current variant, but first two params with underscore

Comment (kamilmielnik) [reply frontend/src/metabase-lib/metadata.ts:160]: "optional" is not an accurate word here. Their presence is necessary, but they can be anything, i.e.
`Lib.displayInfo(null, undefined, query)` will work.

Comment (kamilmielnik) [reply frontend/src/metabase-lib/metadata.ts:160]: > anyhow, I'd not spend time on it, if @kamilmielnik is agreed, I'd go with a current variant, but first two params with underscore

Yeah, even though it bothers me a lot, we can just proceed with the 3 arguments.

To make it even more obvious that these arguments don't matter, we can type them as `unknown`:
```
declare function DisplayInfoFn(
  _query: unknown,
  _stageIndex: unknown,
  query: Query,
): QueryDisplayInfo;
```

Comment (uladzimirdev) [reply frontend/src/metabase-lib/metadata.ts:160]: first parameter has to be a query seems

File: frontend/src/metabase-lib/metadata.ts [hunk 3]
```diff
@@ -143,6 +144,23 @@ declare function DisplayInfoFn(
...
+ *
+ * @example
+ * const query = question.query();
+ * const { isEditable, isNative } = Lib.displayInfo(query, -1, query);
+ *
+ */
+declare function DisplayInfoFn(
+  _query: Query,
+  _stageIndex: number,
+  queryInfo: Query,
```
Comment (uladzimirdev) [inline frontend/src/metabase-lib/metadata.ts:162]: just query
```suggestion
  query: Query,
```

Comment (nemanjaglumac) [reply frontend/src/metabase-lib/metadata.ts:162]: 🤦

Comment (nemanjaglumac) [reply frontend/src/metabase-lib/metadata.ts]: Sorry about that.
Fixed it and removed JSDoc `@comment` in https://github.com/metabase/metabase/pull/37786/commits/265fb7d4826063158af1b680eafd4657ab546fb8

**Review by kamilmielnik — APPROVED** [review]

**Review by uladzimirdev — APPROVED** [review]
:shipit:

**Pull request merged**